src/components/button/button.spec.js

```diff
@@ -0,0 +1,364 @@
+      expect(wrapper.props().disabled).toEqual(false);
+      expect(wrapper).toMatchSnapshot();
+    });
+  });
+
+  describe('when the no props are passed except children', () => {
+    it('sets the default variants to "secondary" and "medium"', () => {
+      const wrapper = render({ children: 'foo' }, TestRenderer.create).toJSON();
+      assertStyleMatch({
+        background: 'transparent',
+        borderColor: BaseTheme.colors.primary,
+        color: BaseTheme.colors.primary,
+        fontSize: '14px',
+        height: '40px',
+        paddingLeft: '24px',
+        paddingRight: '24px'
+      }, wrapper);
+    });
+  });
+
+  describe('when setting the "as" prop', () => {
+    variants.forEach((variant) => {
+      it(`matches the style for a ${variant} Button`, () => {
+        const wrapper = render({
+          children: 'foo', disabled: true, renderAs: variant
+        }, TestRenderer.create).toJSON();
+
+        assertStyleMatch({
+          background:
+          (variant === 'secondary' || variant === 'tertiary' ? 'transparent' : BaseTheme.disabled.button),
+          borderColor: (variant === 'secondary' ? BaseTheme.disabled.button : 'transparent'),
+          color: BaseTheme.disabled.text
+        }, wrapper);
+      });
+    });
+  });
+
+  describe('when the "disabled" prop is passed', () => {
+    it('matches the style for the default Button when no "as" and "size" props are passed', () => {
+      const wrapper = render({ children: 'foo', disabled: true }, TestRenderer.create).toJSON();
+      assertStyleMatch({
+        background: 'transparent',
+        borderColor: BaseTheme.disabled.button,
+        color: BaseTheme.disabled.text,
+        fontSize: '14px',
+        height: `${sizes.medium[0].toString()}px`,
+        paddingLeft: `${sizes.medium[1].toString()}px`,
+        paddingRight: `${sizes.medium[1].toString()}px`
+      }, wrapper);
+    });
+
+    Object.keys(sizes).forEach((size) => {
+      variants.forEach((variant) => {
+        it(`matches the style for a ${size} ${variant} Button`, () => {
+          const wrapper = render({
+            children: 'foo', disabled: true, renderAs: variant, size
+          }, TestRenderer.create).toJSON();
+
+          assertStyleMatch({
+            background:
+            (variant === 'secondary' || variant === 'tertiary' ? 'transparent' : BaseTheme.disabled.button),
+            borderColor: (variant === 'secondary' ? BaseTheme.disabled.button : 'transparent'),
+            color: BaseTheme.disabled.text,
+            fontSize: (size === 'large' ? '16px' : '14px'),
+            height: `${sizes[size][0].toString()}px`,
+            paddingLeft: `${sizes[size][1].toString()}px`,
+            paddingRight: `${sizes[size][1].toString()}px`
+          }, wrapper);
+        });
+      });
+    });
+  });
+
+  describe('when the classic theme is applied', () => {
+    ['blue', 'grey', 'magenta', 'magenta-dull', 'red', 'white'].forEach((variant) => {
```

Comment:
with `describe.each` you could even insert each string into a description using `%s`